src/core/api/quota/registry/registry.go

```diff
@@ -275,11 +280,17 @@ func persistPB(projects []quota.ProjectInfo) error {
 			log.Error(err)
 			return err
 		}
-		_, err = dao.AddBlobsToProject(pro.ProjectID, blobsOfPro...)
-		if err != nil {
-			log.Error(err)
-			return err
+		for _, blobOfPro := range blobsOfPro {
+			_, err = dao.AddBlobToProject(blobOfPro.ID, pro.ProjectID)
+			if err != nil {
+				log.Error(err)
+				if err == dao.ErrDupRows {
+					continue
+				}
+				return err
+			}
```

Comment:
Suggest to use AddBlobsToProject which will be faster than insert blob to project one by one.